jrnl/journals/FolderJournal.py

```diff
@@ -143,15 +148,16 @@ def _get_month_folders(path: pathlib.Path) -> list[pathlib.Path]:
         return
 
     @staticmethod
-    def _get_day_files(path: pathlib.Path) -> list[str]:
-        for child in path.glob(DAY_PATTERN):
+    def _get_day_files(path: pathlib.Path, extension: str) -> list[str]:
+        for child in path.glob(DAY_PATTERN + extension):
```

Comment:
I think the only thing left on this PR is supporting a lack of dots in the extension. For instance, I'd expect this to work the same whether my config value is `extension: md` or `extension: .md`